tophat/jenkins-timeline-plugin
# feat: ported app to React

## Description
This ports the webapp from an inline jQuery snippet to a React app and updates the build scripts to build the webapp first, then the plugin.

## DevQA
### DevQA Prep
### DevQA Steps
- Run `make build_all` to build the plugin and the webapp.
- In Jenkins, use the `Upload` feature to upload your `hpi` package, found in the `/target` directory that is created by the build process.
- Create a pipeline with some steps or use the sample `Jenkinsfile` at the bottom of this description.
- Build your pipeline and check that the `Build timeline` option in the build's page leads you to the timeline.


### Comments:
Sample Jenkinsfile:
```
node {
   def mvnHome
   stage('Preparation') { // for display purposes
    sleep 5
    echo "some text"
    sleep 10
   }
   stage('Build') {
     echo "build build build"
     echo "build build build"
     echo "build build build"
     echo "build build build"
     sleep 2
     echo "build build build"
     echo "build build build"
     echo "build build build"
     echo "build build build"
     echo "build build build"
     sleep 10
     echo "build build build"
     echo "build build build"
     echo "build build build"
     echo "build build build"
     echo "build build build"
     sleep 5
   }
   stage('Results') {
     echo "result result result"
     sleep 3
   }
}
```
